snappy/snap_local.go

```diff
@@ -484,3 +484,137 @@ func (s *SnapPart) RefreshDependentsSecurity(oldPart *SnapPart, inter interacter
 
 	return nil
 }
+
+// SnapPartBag is a slice of *SnapParts that share their name (+origin)
+type SnapPartBag []*SnapPart
+
+// Best SnapPart in the bag (and its index).
```

Comment:
Up next after lightweight (not necessarily right after) is hopefully "part" and "bag". "Best" here is also a pretty bad term, since it doesn't really define what sort of criteria is being used to pick one of the items.

That said, these are future concerns. That logic already existed, and this branch kills lightweight which is a great achievement. Thank you!